word_language_model/model.py

```diff
@@ -1,57 +1,165 @@
+            try:
+                from torch.nn import TransformerEncoder, TransformerEncoderLayer
+            except:
+                raise ImportError('TransformerEncoder module does not exist in PyTorch 1.1 or lower.')
+            if nhead is None:
+                assert self.model_type != 'Transformer'
+            self.src_mask=None
+            self.pos_encoder = PositionalEncoding(ninp, dropout)
+            encoder_layers = TransformerEncoderLayer(ninp, nhead, nhid, dropout)
+            self.model = TransformerEncoder(encoder_layers, nlayers)
         else:
             try:
-                nonlinearity = {'RNN_TANH': 'tanh', 'RNN_RELU': 'relu'}[rnn_type]
+                nonlinearity = {'RNN_TANH': 'tanh', 'RNN_RELU': 'relu'}[self.model_type]
             except KeyError:
                 raise ValueError( """An invalid option for `--model` was supplied,
                                  options are ['LSTM', 'GRU', 'RNN_TANH' or 'RNN_RELU']""")
-            self.rnn = nn.RNN(ninp, nhid, nlayers, nonlinearity=nonlinearity, dropout=dropout)
+            self.model = nn.RNN(ninp, nhid, nlayers, nonlinearity=nonlinearity, dropout=dropout)
         self.decoder = nn.Linear(nhid, ntoken)
 
-        # Optionally tie weights as in:
-        # "Using the Output Embedding to Improve Language Models" (Press & Wolf 2016)
-        # https://arxiv.org/abs/1608.05859
-        # and
-        # "Tying Word Vectors and Word Classifiers: A Loss Framework for Language Modeling" (Inan et al. 2016)
-        # https://arxiv.org/abs/1611.01462
-        if tie_weights:
-            if nhid != ninp:
-                raise ValueError('When using the tied flag, nhid must be equal to emsize')
-            self.decoder.weight = self.encoder.weight
+        if self.model_type == 'Transformer':
+            self._reset_parameters()
+
+        else:
+            # Optionally tie weights as in:
+            # "Using the Output Embedding to Improve Language Models" (Press & Wolf 2016)
+            # https://arxiv.org/abs/1608.05859
+            # and
+            # "Tying Word Vectors and Word Classifiers: A Loss Framework for Language Modeling" (Inan et al. 2016)
+            # https://arxiv.org/abs/1611.01462
+            if tie_weights:
+                if nhid != ninp:
+                    raise ValueError('When using the tied flag, nhid must be equal to emsize')
+                self.decoder.weight = self.encoder.weight
+
+            self.init_weights()
 
-        self.init_weights()
+            self.model_type = model_type
+            self.nhid = nhid
+            self.nlayers = nlayers
 
-        self.rnn_type = rnn_type
-        self.nhid = nhid
-        self.nlayers = nlayers
+    def _reset_parameters(self):
+        r"""Initiate parameters in the model."""
+        assert self.model_type == 'Transformer'
+
+        for p in self.parameters():
+            if p.dim() > 1:
+                nn.init.xavier_uniform_(p)
+
+    def _set_src_mask(self, mask):
```

Comment:
This only has two call sites right next to each other. I think you can inline it and get rid of it.